### PR TITLE
Damienbod/aspnetcore/security/authorization/roles.md update 3.0

### PR DESCRIPTION
**Pull request by damienbod**
Update the role.md to .NET Core 3.0 and maintain the 2.2 code snippets

https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/security/authorization/roles.md

@scottaddie @Rick-Anderson

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@HaoK minor 3.0 update, can you approve?

File: aspnetcore/security/authorization/roles/samples/3_0/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        #region snippet
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<ApplicationDbContext>(options =>
+                options.UseSqlServer(
+					Configuration.GetConnectionString("DefaultConnection")));
+            services.AddIdentity<IdentityUser, IdentityRole>()
+                .AddRoles<IdentityRole>()
```
Comment (serpent5) [inline aspnetcore/security/authorization/roles/samples/3_0/Startup.cs]: I don't think `AddRoles` is needed here as this is already handled using `AddIdentity`, but I do see that the section is about adding this call. Should this just use `AddDefaultIdentity` too, as in the 2.2 version?:

```csharp
services.AddDefaultIdentity<IdentityUser>()
    .AddRoles<IdentityRole>()
    .AddEntityFrameworkStores<ApplicationDbContext>();
```

File: aspnetcore/security/authorization/roles/samples/3_0/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            app.UseHttpsRedirection();
+            app.UseStaticFiles();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllerRoute(
```
Comment (serpent5) [inline aspnetcore/security/authorization/roles/samples/3_0/Startup.cs]: Could this just be `endpoints.MapDefaultControllerRoute();`?

File: aspnetcore/security/authorization/roles/samples/3_0/Startup.cs (hunk 1 above)
Comment (HaoK) [reply aspnetcore/security/authorization/roles/samples/3_0/Startup.cs]: `AddIdentityCore` perhaps?  altho then that requires calling add authentication.  Or maybe remove the call to `AddDefaultUI` instead

Comment (damienbod) [reply aspnetcore/security/authorization/roles/samples/3_0/Startup.cs]: I updated this to the fix from @serpent5 , Is this ok?

File: aspnetcore/security/authorization/roles/samples/3_0/Startup.cs (hunk 2 above)
Comment (damienbod) [reply aspnetcore/security/authorization/roles/samples/3_0/Startup.cs]: fixed

**Comment by Rick-Anderson** [comment]
@HaoK  quick review when you have time, no hurry.

**Pull request merged**